From c574bf79775a3c5af37c2828f30aab2e08d94562 Mon Sep 17 00:00:00 2001
From: Richard Zhu <hongxing.zhu@nxp.com>
Date: Fri, 17 Nov 2017 14:32:23 +0800
Subject: [PATCH 2845/5242] MLK-16818-1 ARM64: dts: imx8: modify the dts to
 enable ep rc support

commit  4a721780771d6579b05bdd462e0164d09c55b0d5 from
https://source.codeaurora.org/external/imx/linux-imx.git

- Correct the comments of iMX8QM PCIEB
- Enlarge the CFG space of iMX8QXP PCIEB.
- PCIE port maybe hard-wired in the hardware design.
Use the hard-wired property to specify it on iMX8MQ.

Signed-off-by: Richard Zhu <hongxing.zhu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../devicetree/bindings/pci/fsl,imx6q-pcie.txt     |    1 +
 arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts   |    3 +++
 arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi      |    2 +-
 arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi     |    2 +-
 4 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/Documentation/devicetree/bindings/pci/fsl,imx6q-pcie.txt b/Documentation/devicetree/bindings/pci/fsl,imx6q-pcie.txt
index 30c8e9f..b9212e3 100644
--- a/Documentation/devicetree/bindings/pci/fsl,imx6q-pcie.txt
+++ b/Documentation/devicetree/bindings/pci/fsl,imx6q-pcie.txt
@@ -20,6 +20,7 @@ Required properties:
 - clock-names: Must include the following additional entries:
 	- "pcie_phy"
 - ext_osc: use the external oscillator or not.
+- hard-wired: the port is hard wired in hw design or not.
 
 Optional properties:
 - fsl,tx-deemph-gen1: Gen1 De-emphasis value. Default: 0
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts
index fcc0bf1..82e674c 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts
@@ -472,6 +472,8 @@
 	clkreq-gpio = <&gpio5 20 GPIO_ACTIVE_LOW>;
 	disable-gpio = <&gpio5 29 GPIO_ACTIVE_LOW>;
 	reset-gpio = <&gpio5 28 GPIO_ACTIVE_LOW>;
+	ext_osc = <1>;
+	hard-wired = <1>;
 	status = "okay";
 };
 
@@ -481,6 +483,7 @@
 	clkreq-gpio = <&gpio5 21 GPIO_ACTIVE_LOW>;
 	disable-gpio = <&gpio5 10 GPIO_ACTIVE_LOW>;
 	reset-gpio = <&gpio5 12 GPIO_ACTIVE_LOW>;
+	ext_osc = <1>;
 	status = "okay";
 };
 
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi
index 91b4926..e10b43c 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi
@@ -3019,7 +3019,7 @@
 		fsl,max-link-speed = <3>;
 		hsio-cfg = <PCIEAX1PCIEBX1SATA>;
 		hsio = <&hsio>;
-		ctrl-id = <1>; /* pciea */
+		ctrl-id = <1>; /* pcieb */
 		cpu-base-addr = <0x80000000>;
 		status = "disabled";
 	};
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
index 4ec70d5..8872d3d 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
@@ -2240,7 +2240,7 @@
 		 */
 		compatible = "fsl,imx8qxp-pcie","snps,dw-pcie";
 		reg = <0x0 0x5f010000 0x0 0x10000>, /* Controller reg*/
-		      <0x0 0x7ff00000 0x0 0x10000>; /* PCI cfg space */
+		      <0x0 0x7ff00000 0x0 0x80000>; /* PCI cfg space */
 		reg-names = "dbi", "config";
 		#address-cells = <3>;
 		#size-cells = <2>;
-- 
1.7.9.5

